src/Gateways/GpApiGateway.php

```diff
@@ -376,6 +376,23 @@ protected function add_hooks() {
 			'process_threeDSecure_challengeNotification'
 		) );
 		add_action( 'woocommerce_order_actions', array( $this, 'handle_adding_capture_order_action' ) );
+
+		add_action('woocommerce_scheduled_subscription_payment_' . $this->id, array($this, "renew_subscription"), 10, 3);
+		// When subscription is canceled remove the meta data
+		add_action('woocommerce_subscription_status_pending-cancel_to_cancelled', function($subscription){
+			if($subscription->get_payment_method() === $this->id){
```

Comment:
This if/else block seems to be incorrectly indented. Possibly due to a combination of spaces and tabs. We should use only tabs. You should be able to configure your coding editor to show the spaces/tabs.
The white spaces comment applies in here too.